### PR TITLE
Non linearity after attention

### LINKED ISSUE #3: Use of non-linearity in calculating attention
Are you sure there is no tanh/relu in line 47 of  seq2seq/model.py? If I understand it correctly, this is a one-layer NN, so shouldn't there be a non-linearity in between?

### PR DESCRIPTION
**Pull request by pskrunner14**
Adds non-linearity after the computation of attention weights. Fixes #3

### CODE REVIEW

**Comment by keon** [comment]
thanks so much :)

**Comment by CuriousG102** [comment]
F.relu does not have a kwarg dim?

**Comment by CuriousG102** [comment]
Not only does this not run until you remove dim=1 from the ReLU, this softmax does not make sense. It computes softmax along the batch dimension...

File: model.py [hunk 1]
```diff
@@ -40,11 +40,11 @@ def forward(self, hidden, encoder_outputs):
         h = hidden.repeat(timestep, 1, 1).transpose(0, 1)
         encoder_outputs = encoder_outputs.transpose(0, 1)  # [B*T*H]
         attn_energies = self.score(h, encoder_outputs)
-        return F.softmax(attn_energies, dim=1).unsqueeze(1)
+        return F.relu(attn_energies, dim=1).unsqueeze(1)
 
     def score(self, hidden, encoder_outputs):
         # [B*T*2H]->[B*T*H]
-        energy = self.attn(torch.cat([hidden, encoder_outputs], 2))
+        energy = F.softmax(self.attn(torch.cat([hidden, encoder_outputs], 2)))
```
Comment (CuriousG102) [inline model.py:47]: I believe you need a ReLU here.

File: model.py [hunk 2]
```diff
@@ -40,11 +40,11 @@ def forward(self, hidden, encoder_outputs):
         h = hidden.repeat(timestep, 1, 1).transpose(0, 1)
         encoder_outputs = encoder_outputs.transpose(0, 1)  # [B*T*H]
         attn_energies = self.score(h, encoder_outputs)
-        return F.softmax(attn_energies, dim=1).unsqueeze(1)
+        return F.relu(attn_energies, dim=1).unsqueeze(1)
```
Comment (CuriousG102) [inline model.py:43]: And to leave this line alone
